### PR TITLE
Issue #416

### PR DESCRIPTION
**Pull request by AminDhouib**
Fixing this issue: https://github.com/curveball/a12n-server/issues/416 (All Steps)

### CODE REVIEW

File: src/oauth2-client/controller/new.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ class NewClientController extends Controller {
     }
 
     ctx.response.type = 'application/vnd.siren+json';
-    ctx.response.body = siren.newClient(user);
+    ctx.response.body = siren.newClient(user, ctx.query.client_id);
```
[the remark this answers is not in the document]
Comment (AminDhouib) [reply src/oauth2-client/controller/new.ts:18]: Got it

[the remark this answers is not in the document]
Comment (AminDhouib) [reply src/oauth2-client/controller/new.ts]: Done. Here is an example of how I have defined the queries:
http://localhost:8531/app/wchU5JBFaTo/client/new?**redirectUris**=https://a.com&**redirectUris**=https://b.com&**allowAuthorizationCode**&**clientId**=TestClientName

File: src/oauth2-client/formats/siren.ts [hunk 2]
```diff
@@ -1,9 +1,14 @@
 import { App } from '../../principal/types';
 
-export function newClient(user: App, clientIdValue: string) {
+export function newClient(user: App, query: any ) {
```
Comment (evert) [inline src/oauth2-client/formats/siren.ts:3]: Could you write a type for this?

**Comment by AminDhouib** [comment]
This latest PR includes all of the steps done (Steps 1-3) from Issue #416.

**Review by mhum — COMMENTED** [review]
Overall, pretty good! Mostly just comments about conventions we use and some syntax recommendations.

**Comment by AminDhouib** [comment]
Applied all of your awesome feedback @mhum let me know what you think :)

File: schemas/app-new-form.json [hunk 3]
```diff
@@ -25,6 +25,18 @@
...
+      "type": "string"
+    },
+    "allowGrantTypes": {
+      "type": "string"
+    },
+    "redirectUris": {
+      "type": "string"
+    },
+    "requirePkce": {
+      "type": "string"
```
Comment (mhum) [inline schemas/app-new-form.json:39]: Can this be a `boolean`? There are a handful of places in this PR that it's typed as a `string`, but can it be coerced to a boolean at some point so the typing is more accurate?

File: src/oauth2-client/formats/siren.ts [hunk 4]
```diff
@@ -1,9 +1,20 @@
...
+ allowGrantTypes?: string;
+ redirectUris?: string;
+ requirePkce?: string;
+}
+
+export function newClient(user: App, query: NewClientQuery ) {
 
   const memberHref = `/app/${user.id}`;
 
+  const redirectUris = query.redirectUris ? (query.redirectUris + '').split(',').join('\n') : '';
```
Comment (mhum) [inline src/oauth2-client/formats/siren.ts:14]: Why the string concatenation here and for `allowGrantTypes`? Are these coming in as a different type and you're convert them to strings?

Comment (AminDhouib) [reply src/oauth2-client/formats/siren.ts:14]: Yeah, without them I get a `TypeError: query.redirectUris.split is not a function`
Any better ways you would suggest to eliminate the error perhaps?

**Review by mhum — APPROVED** [review]

**Pull request merged**